tests/components/fritz/test_config_flow.py

```diff
@@ -0,0 +1,361 @@
+        side_effect=fc_class_mock,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_SSDP}, data=MOCK_SSDP_DATA
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "confirm"
+
+        MOCK_NO_UNIQUE_ID = MOCK_SSDP_DATA.copy()
+        del MOCK_NO_UNIQUE_ID[ATTR_UPNP_UDN]
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_SSDP}, data=MOCK_NO_UNIQUE_ID
+        )
+        assert result["type"] == RESULT_TYPE_ABORT
+        assert result["reason"] == "already_in_progress"
+
+
+async def test_ssdp(hass: HomeAssistantType, fc_class_mock):
+    """Test starting a flow from discovery but no device found."""
+    with patch(
+        "homeassistant.components.fritz.common.FritzConnection",
+        side_effect=fc_class_mock,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_SSDP}, data=MOCK_SSDP_DATA
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "confirm"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_USERNAME: "fake_user",
+                CONF_PASSWORD: "fake_pass",
+            },
+        )
+
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result["data"][CONF_HOST] == "fake_host"
+        assert result["data"][CONF_PASSWORD] == "fake_pass"
+        assert result["data"][CONF_USERNAME] == "fake_user"
+
+
+async def test_ssdp_exception(hass: HomeAssistantType):
+    """Test starting a flow from discovery but no device found."""
+    with patch(
+        "homeassistant.components.fritz.common.FritzConnection",
+        side_effect=FritzConnectionException,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_SSDP}, data=MOCK_SSDP_DATA
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "confirm"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_USERNAME: "fake_user",
+                CONF_PASSWORD: "fake_pass",
+            },
+        )
+
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "confirm"
+
+
+async def test_import(hass: HomeAssistantType):
+    """Test importing."""
+
+    flow = config_flow.FritzBoxToolsFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_import(MOCK_IMPORT_CONFIG)
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "start_config"
```

Comment:
Finish the test all the way though and verify the end result is `RESULT_TYPE_CREATE_ENTRY` with the imported data

Reply:
Done